Feat assignment 7 git in visual studio (#8)
aristovinceselvanathan/TIPS: src/Assignment7/Git/Readme.md

# Building a Simple Project with Git in Visual Studio
## Objective: 
The goal of this assignment is to familiarize yourself with the Git functionalities within Visual
Studio. You will create a simple project, make changes, and use Git for version control.
### Part I: Setting Up Your Environment
1. Install Git and Visual Studio, if you haven't already.

> 2. Open Visual Studio and createa new project. Choose a simple console application for this assignment.<br><br>

Reviewer: ```suggestion
2. Open Visual Studio and create a new project. Choose a simple console application for this assignment.<br><br>
```

Author: I have changed the statement

> ![Alt text](<Screenshot (13).png>)

Reviewer: Please update the name of the screenshots such that it describes what is shown.

Author: I have resolved it

### Part 2: Initializing a Git Repository
1. Initialize a new Git repository for your project in Visual Studio.
2. Make an initial commit with the project's starting code.<br><br>
![Alt text](<Screenshot (2).png>)
### Part 3: Making and Staging Changes
1. Make some changes to your code. This could be adding a new feature or modifying an existing one.
2. Stage these changes using the 'Changes' window in Visual Studio.<br><br>
![Alt text](<Screenshot (5).png>)
### Part 4: Committing Changes

> 1. Writea meaningful commit message describing the changes you made.

Reviewer: ```suggestion
1. Write a meaningful commit message describing the changes you made.
```

Author: I Have changed the statement

2. Commit your changes.<br><br>
![Alt text](<Screenshot (6).png>)
### Part 5: Branching and Merging
1. Create a new branch and switch to it.
2. Make some changes in this new branch and commit them. 
3. Switch back to the main branch and merge the changes from the new branch.<br><br>
![Alt text](<Screenshot (7).png>)
### Part 6: Resolving Merge Conflicts
1. Create a situation where a merge conflict would occur (e.g., make conflicting changes in two branches).
2. Attempt to merge the branches and resolve the resulting conflict.<br><br>
![Alt text](<Screenshot (16).png>)<br>
![Alt text](<Screenshot (15).png>)<br>
![Alt text](<Screenshot (17).png>)<br>
### Part 7: Viewing Git History
1. View the commit history of your repository.
2. Identify the changes made in each commit.<br><br>
![Alt text](<Screenshot (8).png>)
## Part 8: Selective Staging
1. Make several changes to a file in your project.
2. Use selective staging to stage only some of these changes and commit them.
3. Stage and commit the remaining changes separately.<br><br>

> ![Alt text](<Screenshot (10).png>)<br>

Reviewer: It's good that you've added all these screenshots for reference.

# Documentation Report
## Overview

> The assignemnt that can able to understand the concept of the git in Visual Studio by the use of the git chages window. Became familar with the window.

Reviewer: Please update this statement.

Author: I have updated the statement

## Requirements
1. Git Repository
2. Visual Studio
3. Project Files (Git Files)

## Description about the methods in Program Class:

> Main method just prints the required informations like Hello World, How are you etc.

Reviewer: ```suggestion
The main method just prints the required information like Hello World, How are you, etc.
```

Author: I have changed the statement

## Implementation Considerations

> - Resolving the merge confilct between the branches

Reviewer: ```suggestion
- Resolving the merge conflicts between the branches
```

Author: I have changed the statement

- Staging the required files

> - Creating the new branch from the current branch

Reviewer: ```suggestion
- Creating a new branch from the current branch
```

Author: I have changed the statement
